deriv_api/deriv_api.py

```diff
@@ -246,14 +249,41 @@ async def send(self, request: dict) -> dict:
         -------
             API response
         """
+
+        send_will_be_called = self._call_middleware('sendWillBeCalled', {'request': request})
+        if send_will_be_called:
+            return send_will_be_called
+
         response_future = self.send_and_get_source(request).pipe(op.first(), op.to_future())
 
         response = await response_future
         self.cache.set(request, response)
         if self.storage:
             self.storage.set(request, response)
+        send_is_called = self._call_middleware('sendIsCalled',{'response': response, 'request': request})
+        if send_is_called:
+            return send_is_called
         return response
 
+    def _call_middleware(self, name: str, args: dict) -> Union[None,dict]:
```

Comment:
refactor it to the separate class

Reply:
How ? Could you please give me soem psudo-code ?

Reply:
I mean write a new middleware class and add this method there

Reply:
I guess like this:
```
class Middleware:
    def __init__(self, name, code):
        self.name = name
        self.code = code

```
And in deriv_api class:

Reply:
middlewares are useful when used stack-like manner, allowing you to perform actions downstream then filter and manipulate the response upstream. This means in a small workflow like this is not much need but keeping in mind the future, its okay